### PR TITLE
DSD-509: Chakra HelperErrorText

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-509](https://jira.nypl.org/browse/DSD-509)

## This PR does the following:
- Updates the `HelperErrorText` component with Chakra.

## How has this been tested?

Storybook.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/HelperErrorText/HelperErrorText.tsx [hunk 1]
```diff
@@ -28,36 +31,27 @@ export default function HelperErrorText(
...
-    <div
+    <Box
       id={id}
-      className={bem(baseClass, modifiers, blockName)}
-      aria-live={announceAriaLive ? ariaLive : "off"}
+      className={className}
       aria-atomic={ariaAtomic}
-      {...attributes}
+      data-isInvalid={isInvalid}
+      aria-live={announceAriaLive ? ariaLive : "off"}
```
Comment (bigfishdesign13) [inline src/components/HelperErrorText/HelperErrorText.tsx:51]: We don't use this in many of the other components.  Why is that?

File: src/theme/components/helperErrorText.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const helperErrorText = {
+  baseStyle: ({ isInvalid }) => ({
+    marginBottom: "xs",
```
Comment (bigfishdesign13) [inline src/theme/components/helperErrorText.ts]: The original component did have `bottom-margin` set to `xs`, but I think this should be removed and the bottom-margin should be 0.  The `Form` component will **now** handle the spacing between form elements.  Once this is set to 0, we will need to review the other components visually to make sure this margin value was not be relied upon.  Off the top of my head, I think `DatePicker` may be the only one that really needs it.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
The requested changes tie into what I was saying in the PR for DSD-523.

File: src/components/HelperErrorText/HelperErrorText.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/HelperErrorText/HelperErrorText.tsx:51]: Do you mean the `aria-live` attribute itself? I think it fits best for this type of component where the description test might change (often) and is useful to the user and isn't already being presented (such as through a label or link).

**Comment by EdwinGuzman** [comment]
I added the changes from DSD-523 PR #721  in the last commit for this PR. I also updated the `Form` stories so it displays the `DatePicker` component as an example.

File: src/components/HelperErrorText/HelperErrorText.tsx (hunk 1 above)
Comment (bigfishdesign13) [reply src/components/HelperErrorText/HelperErrorText.tsx:51]: Got it.  Makes sense.  Thanks.

File: src/theme/components/helperErrorText.ts (hunk 2 above)
Comment (bigfishdesign13) [reply src/theme/components/helperErrorText.ts]: I had a typo in this comment.  It should have been...
```The Form component will NOW handle the spacing...```

**Review by bigfishdesign13 — APPROVED** [review]
This looks good.  We just need to make sure to fix the spacing issues now present in `DatePicker`.

**Pull request merged**